osde2e/route_monitor_operator_tests.go

```diff
@@ -0,0 +1,235 @@
+//go:build osde2e
+// +build osde2e
+
+package osde2etests
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/osde2e-common/pkg/clients/openshift"
+	. "github.com/openshift/osde2e-common/pkg/gomega/assertions"
+	. "github.com/openshift/osde2e-common/pkg/gomega/matchers"
+	routemonitorv1alpha1 "github.com/openshift/route-monitor-operator/api/v1alpha1"
+	prometheusop "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	corev1 "k8s.io/api/core/v1"
+	kubev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = Describe("Route Monitor Operator", Ordered, func() {
+	var (
+		k8s      *openshift.Client
+		err      error
+		pod      *corev1.Pod
+		svc      *corev1.Service
+		appRoute *routev1.Route
+		rmo      *routemonitorv1alpha1.RouteMonitor
+	)
+	const (
+		operatorNamespace            = "openshift-route-monitor-operator"
+		deploymentName               = "route-monitor-operator-controller-manager"
+		operatorName                 = "route-monitor-operator"
+		consoleName                  = "console"
+		pollingDuration              = 10 * time.Minute
+		routeMonitorName             = "routemonitor-e2e-test"
+		defaultDesiredReplicas int32 = 1
+	)
+	BeforeAll(func(ctx context.Context) {
+		log.SetLogger(GinkgoLogr)
+		k8s, err = openshift.New(GinkgoLogr)
+		Expect(err).ShouldNot(HaveOccurred(), "unable to setup k8s client")
+		Expect(routemonitorv1alpha1.AddToScheme(k8s.GetScheme())).Should(Succeed(), "unable to register RouteMonitor scheme")
+	})
+
+	It("is installed", func(ctx context.Context) {
+		By("checking the deployment exists and is available")
+		EventuallyDeployment(ctx, k8s, deploymentName, operatorNamespace).Should(BeAvailable())
+	})
+
+	It("can be upgraded", func(ctx context.Context) {
+		By("forcing operator upgrade")
+		err = k8s.UpgradeOperator(ctx, operatorName, operatorNamespace)
+		Expect(err).NotTo(HaveOccurred(), "operator upgrade failed")
+	})
+
+	It("has all of the required resources", func(ctx context.Context) {
+		promclient, err := prometheusop.NewForConfig(k8s.GetConfig())
+		Expect(err).ShouldNot(HaveOccurred(), "failed to configure Prometheus-operator clientset")
+		_, err = promclient.MonitoringV1().ServiceMonitors(operatorNamespace).Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "Could not get console serviceMonitor")
+		_, err = promclient.MonitoringV1().PrometheusRules(operatorNamespace).Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "Could not get console prometheusRule")
+	})
+
+	It("Creates and deletes a RouteMonitor to see if it works accordingly", func(ctx context.Context) {
```

Comment:
This test name should be something like `required dependent resources are created` because when you create a RouteMonitor resource, the service monitor and prometheus rules are created for the RM